public/app/features/dashboard/state/DashboardMigrator.test.ts

```diff
@@ -1242,6 +1242,156 @@ describe('DashboardModel', () => {
       `);
     });
   });
+
+  describe('when migrating singlestat value mappings', () => {
+    it('should migrate value mapping', () => {
+      const model = new DashboardModel({
+        panels: [
+          {
+            type: 'singlestat',
+            legend: true,
+            thresholds: '10,20,30',
+            colors: ['#FF0000', 'green', 'orange'],
+            aliasYAxis: { test: 2 },
+            grid: { min: 1, max: 10 },
+            targets: [{ refId: 'A' }, {}],
+            mappingType: 1,
+            mappingTypes: [
+              {
+                name: 'value to text',
+                value: 1,
+              },
+            ],
+            valueMaps: [
+              {
+                op: '=',
+                text: 'test',
+                value: '20',
+              },
+              {
+                op: '=',
+                text: 'test1',
+                value: '30',
+              },
+              {
+                op: '=',
+                text: '50',
+                value: '30',
+              },
+            ],
+          },
+        ],
+      });
+      expect(model.panels[0].fieldConfig.defaults.mappings).toMatchInlineSnapshot(`
+        Array [
+          Object {
+            "options": Object {
+              "20": Object {
+                "color": undefined,
+                "text": "test",
+              },
+            },
+            "type": "value",
+          },
+          Object {
+            "options": Object {
+              "30": Object {
+                "color": undefined,
+                "text": "test1",
+              },
+            },
+            "type": "value",
+          },
+          Object {
+            "options": Object {
+              "30": Object {
+                "color": "orange",
+                "text": "50",
+              },
+            },
+            "type": "value",
+          },
+        ]
+      `);
```

Comment:
this does not look correct, value mappings should be all in one object map

Reply:
maybe that migration happens after this?

Reply:
You are right. Trying to migrate this dashboard:

```
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": "-- Grafana --",
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "gnetId": null,
  "graphTooltip": 0,
  "id": 465,
  "links": [],
  "panels": [
    {
      "cacheTimeout": null,
      "colorBackground": false,
      "colorValue": false,
      "colors": [
        "#299c46",
        "rgba(237, 129, 40, 0.89)",
        "#d44a3a"
      ],
      "datasource": null,
      "format": "none",
      "gauge": {
        "maxValue": 100,
        "minValue": 0,
        "show": false,
        "thresholdLabels": false,
        "thresholdMarkers": true
      },
      "gridPos": {
        "h": 9,
        "w": 12,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "interval": null,
      "links": [],
      "mappingType": 1,
      "mappingTypes": [
        {
          "name": "value to text",
          "value": 1
        },
        {
          "name": "range to text",
          "value": 2
        }
      ],
      "maxDataPoints": 100,
      "nullPointMode": "connected",
      "nullText": null,
      "postfix": "",
      "postfixFontSize": "50%",
      "prefix": "",
      "prefixFontSize": "50%",
      "rangeMaps": [
        {
          "from": "null",
          "text": "N/A",
          "to": "null"
        }
      ],
      "sparkline": {
        "fillColor": "rgba(31, 118, 189, 0.18)",
        "full": false,
        "lineColor": "rgb(31, 120, 193)",
        "show": false,
        "ymax": null,
        "ymin": null
      },
      "tableColumn": "",
      "targets": [
        {
          "refId": "A",
          "scenarioId": "csv_metric_values",
          "stringInput": "90"
        }
      ],
      "thresholds": "",
      "timeFrom": null,
      "timeShift": null,
      "title": "Panel Title",
      "type": "singlestat",
      "valueFontSize": "80%",
      "valueMaps": [
        {
          "op": "=",
          "text": "test",
          "value": "24.3"
        },
        {
          "op": "=",
          "text": "91",
          "value": "190"
        },
        {
          "op": "=",
          "text": "test",
          "value": "90"
        }
      ],
      "valueName": "avg"
    }
  ],
  "schemaVersion": 22,
  "style": "dark",
  "tags": [],
  "templating": {
    "list": []
  },
  "time": {
    "from": "now-6h",
    "to": "now"
  },
  "timepicker": {
    "refresh_intervals": [
      "5s",
      "10s",
      "30s",
      "1m",
      "5m",
      "15m",
      "30m",
      "1h",
      "2h",
      "1d"
    ]
  },
  "timezone": "",
  "title": "test legacy stat",
  "uid": "QYrYmggnz",
  "variables": {
    "list": []
  },
  "version": 1
}
```

results in panel having this field config for mappings:

```
"mappings": [
        {
          "type": "value",
          "options": {
            "24.3": {
              "text": "test"
            }
          }
        },
        {
          "type": "value",
          "options": {
            "190": {
              "text": "91"
            }
          }
        },
        {
          "type": "value",
          "options": {
            "90": {
              "text": "test"
            }
          }
        }
      ],
```

Yet, it seems to be working fine 🤔

Reply:
@torkelo the migration is now updated and collects val->text mappings in a single object. I've added it in the value mappings migration instead of base single stat migration to avoid code duplication.